src/main/java/ultilities/GMail.java

```diff
@@ -0,0 +1,284 @@
+            return !messages.isEmpty();
+        } catch (Exception e) {
+            System.out.println("Exception log" + e);
+            return false;
+        }
+    }
+
+
+    private static List<Message> getMessages(ListMessagesResponse response) {
+        List<Message> messages = new ArrayList<Message>();
+        try {
+            final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+            Gmail service = new Gmail.Builder(HTTP_TRANSPORT, JSON_FACTORY, getCredentials(HTTP_TRANSPORT))
+                    .setApplicationName(APPLICATION_NAME)
+                    .build();
+            while (response.getMessages() != null) {
+                messages.addAll(response.getMessages());
+                if (response.getNextPageToken() != null) {
+                    String pageToken = response.getNextPageToken();
+                    response = service.users().messages().list(USER_ID)
+                            .setPageToken(pageToken).execute();
+                } else {
+                    break;
+                }
+            }
+            return messages;
+        } catch (Exception e) {
+            System.out.println("Exception log " + e);
+            return messages;
+        }
+    }
+
+
+    /**
+     * Get the latest unread message of given to_user_email_id and Subject
+     * Try to find the mail for 10 times
+     * If that email in not found then return null string
+     * @param toEmail
+     * @param subject
+     * @return
+     */
+    public static String getEmailBody(String toEmail, String subject) {
+        String query = "to:("+toEmail+") subject:("+subject+")";
+
+        String body = null;
+
+        for(int i = 0; i<3; i++) {
+            try {
+                body=getGmailData(query).get("body");
+                break;
+            }
+            catch (NullPointerException n) {
+                try {
+                    java.lang.Thread.sleep(1000);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+            catch (Exception e) {
+                e.printStackTrace();
+                break;
+            }
+        }
+        return body;
+
+    }
+
+    public static String getEmailLink(String toEmail, String subject) {
+        String query = "to:("+toEmail+") subject:("+subject+")";
+
+        String link = null;
+
+        for(int i = 0; i<3; i++) {
+            try {
+                link=getGmailData(query).get("link");
+                break;
+            }
+            catch (NullPointerException n) {
+                try {
+                    java.lang.Thread.sleep(1000);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+            catch (Exception e) {
+                e.printStackTrace();
+                break;
+            }
+        }
+        return link;
+
+    }
+
+    /*
```

Comment:
Remove redundant comments code from line 272 to line 282

Reply:
Already done.